sha256_mb/sha256_mb_x16_avx512.asm

```diff
@@ -331,12 +333,12 @@ FIELD	_rsp,		8,	8
 	;; H becomes T2, then add T1 for A
 	;; D becomes D + T1 for E
 
-	vpaddd		T1, H, TMP3		; T1 = H + Kt
+	vpaddd		T1, H, [TBL+%%ROUND*4]{1to16}	; T1 = H + Kt
```

Comment:
I'm not sure that the code would work without the embedded broadcast. With a single value in memory for each constant we need to replicate it onto the full register. Without the {1to16} we'll end up loading a bunch of different constants in each vector element. The advantage of the embedded broadcast  (beside not having to replicate 16 times each constant) is that is basically "free" as far as CPU resources utilization, according to section 18.9.2 of the optimization guide.

Reply:
I don't think that it will work without the <code>{1to16}</code> bit set. 
Without that we'll end up loading a bunch of different scalar constats in each vector element instead of replicating the same constant in all the elements. I get all sorts of tests errors without that. Let me know if i'm missing something 😄